### ISSUE TITLE
Hot field Code Cleanup and Possible Removal of Old Hot Field Features

### ISSUE THREAD

**Issue by jonoommen** [eclipse/omr#5622]
With dynamic breadth first scan ordering in place, there is old hot field related code that should be removed. While doing this, based off an instinct of @charliegracie from a year ago, I tested the removal of hot field alignment and tenured alignment code as well, and as suspected, although hot field padding and tenured alignment at one point must of provided a performance benefit, this appears to be no longer the case. As I have been informed, the feature was enabled upon its positive impacts to SpecJbb2005, but my performance testing on various machines has confirmed this to be no longer the case. I have spoken previously and briefly with @amicic and @andrewcraik regarding this but it has not been formalized until now. I have PR's completed for this and have testing completed as well (please see attached results file). If there are benefits of hot field alignment and tenured alignment I am not aware of or if any additional testing needs to be done to prove this please share.

If you are interested in what the omr side of the change that was tested would look like: 
[https://github.com/eclipse/omr/compare/master...jonoommen:RemoveOldHotFieldRemantsUpdated?expand=1](url)

**Comment by jonoommen** [comment]
[results-summary.pdf](https://github.com/eclipse/omr/files/5410410/results-summary.pdf)

**Comment by jonoommen** [comment]
[results-summary.pdf](https://github.com/eclipse/omr/files/5599660/results-summary.pdf)

**Comment by amicic** [comment]
Hot field alignment is a source of holes in copy-scan caches, which makes the object iteration a bit more complex. 2+% slower Scavenge times. 

From GC perspective, this is still something we want to do.

@vijaysun-omr do you think we want to re-do the testing (and if so, what)?

**Comment by vijaysun-omr** [comment]
Maybe SPECjbb2015 and the Dacapo benchmark suite

**Comment by vijaysun-omr** [comment]
@zl-wang should we try a SPECjbb2015 run without this hot field ordering on the most relevant benchmark setups ? In today's setting, I would expect we can remove this feature if it does not benefit SPECjbb2015 since I don't think we have a more likely scenario that could have benefited.

**Comment by zl-wang** [comment]
yes, giving jbb2015 a try to decide is a reasonable approach going forward.
